Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/utils/StartupHandler.java

```diff
@@ -33,27 +37,60 @@ public static void HandleInit(FragmentActivity parent)
     if (TvUtil.isLeanback(parent))
       TvUtil.scheduleSyncingChannel(parent);
 
-    String[] start_files = null;
-    Bundle extras = parent.getIntent().getExtras();
-    if (extras != null)
+    String[] games_to_launch = getGamesFromIntent(parent.getIntent());
+    if (games_to_launch != null && games_to_launch.length > 0)
     {
-      start_files = extras.getStringArray("AutoStartFiles");
-      if (start_files == null)
+      // Start the emulation activity, send the ISO passed in and finish the main activity
+      EmulationActivity.launch(parent, games_to_launch, false);
+      parent.finish();
+    }
+  }
+
+  private static String[] getGamesFromIntent(Intent intent)
+  {
+    // Priority order when looking for game paths in an intent:
+    //
+    // Specifying multiple discs (for multi-disc games) is prioritized over specifying a single
+    // disc. But most of the time, only a single disc will have been specified anyway.
+    //
+    // Specifying content URIs (compatible with scoped storage) is prioritized over raw paths.
+    // The intention is that if a frontend app specifies both a content URI and a raw path, newer
+    // versions of Dolphin will work correctly under scoped storage, while older versions of Dolphin
+    // (which don't use scoped storage and don't support content URIs) will also work.
+
+    // 1. Content URI, multiple
+    ClipData clipData = intent.getClipData();
+    if (clipData != null)
+    {
+      String[] uris = new String[clipData.getItemCount()];
+      for (int i = 0; i < uris.length; i++)
       {
-        String start_file = extras.getString("AutoStartFile");
-        if (!TextUtils.isEmpty(start_file))
-        {
-          start_files = new String[]{start_file};
-        }
+        uris[i] = Objects.toString(clipData.getItemAt(i).getUri());
```

Comment:
Hi, Here's an idiomatic form that should work on API level 24+
```
  return IntStream.range(0, clipData.getItemCount()).boxed()
          .map(ClipData::::getItemAt)
          .collect(Collectors.toList())
          .toArray(String[]::new);
```

Proof / sample:
```
import java.util.Arrays;
import java.util.List;
import java.util.stream.Collectors;
import java.util.stream.IntStream;

public class Main {

    public static void main(String[] args) {
        List<String> test = List.of("mario", "rosalina", "luigi", "bowser");

        // You would return this
        String[] result = IntStream.range(0, test.size()).boxed()
                .map(test::get)
                .collect(Collectors.toList())
                .toArray(String[]::new);

        // Just to see the output
        Arrays.stream(result).forEach(System.out::println);
    }
}
```

Reply:
Appears minimum is 21 (Android 5 / 2014) though in build.gradle . Disregard if not posssible

Reply:
I will skip this due to the minSdkVersion.